### ISSUE TITLE
Support immersive mode ("real" fullscreen) on Android

### ISSUE THREAD

**Issue by dos1** [liballeg/allegro5#800]
Android supports (since API level 19, that is Android 4.4) a special mode called [immersive](https://developer.android.com/training/system-ui/immersive.html), which hides everything, even navigation buttons on devices without hardware ones, and gives the application access to whole screen. User can then slide out status and navigation bars with slide gesture.

I don't know how to proceed with implementing it in Allegro (should it get a new, platform-specific display flag, or should it be just ALLEGRO_WINDOW_FULLSCREEN, since it's actually just that, a real fullscreen mode? or maybe even just hardcoded default, like standard fullscreen right now? should the change of current behaviour be considered a breaking change then?), so I'll just leave here this code which works for me for you to use :)

```java
if (android.os.Build.VERSION.SDK_INT >= 19) {
   this.getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION);
}
```

at the end of `onCreate` in `android/allegro_activity/src/AllegroActivity.java`, with `import android.view.View;` added earlier.

The downside of this code is that it needs SDK 19 to build, but if that's any issue then I think that might be fixable using reflection. Doesn't change any runtime dependency though.

If adding it just like that is fine for you, I'll be happy to provide a pull request :)

**Comment by SiegeLord** [comment]
We generally don't mind upping the build requirement as long as the end-user requirement is low. if already have a few of such version checks.

**Comment by dos1** [comment]
Fixed by #1071
